### ISSUE TITLE
SAM Testing of lambda - fails while deploying to AWS works!

### ISSUE THREAD

**Issue by javapapo** [micronaut-projects/micronaut-aws#13]
Hello I was going through the article [here](https://guides.micronaut.io/micronaut-function-aws-lambda/guide/index.html) .

I decided to experiment with a far simpler version - following though the steps of the article.
Full example project:https://www.dropbox.com/s/af87ua716lh85cw/papolambda.zip

Practically is a simple λ function that returns a String nothing special.

I have tried to test it with `AWS ASM`
```bash
sam --version
SAM CLI, version 0.11.0

echo '{}' | sam local invoke papolambda
```

```xml
  <properties>
    <micronaut.version>1.1.0.M1</micronaut.version>
    <jdk.version>8</jdk.version>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    <exec.mainClass>io.micronaut.function.executor.FunctionApplication</exec.mainClass>
  </properties>
```
using the template.yaml below  : 

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Globals:
  Function:
    Timeout: 25
Resources:
  papolambda:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: target/papolamda-0.1.jar
      Handler: io.micronaut.function.aws.MicronautRequestStreamHandler
      Runtime: java8
```

Where I get the following error :

```bash
START RequestId: f778d699-f29b-4410-a211-8ee626e6fb90 Version: $LATEST
java.lang.ClassNotFoundException: io.micronaut.function.aws.MicronautRequestStreamHandler
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)

END RequestId: f778d699-f29b-4410-a211-8ee626e6fb90
REPORT RequestId: f778d699-f29b-4410-a211-8ee626e6fb90	Duration: 6.81 ms	Billed Duration: 100 ms	Memory Size: 128 MB	Max Memory Used: 3 MB
```

While deploying to AWS it works!

**Comment by mengjiann** [comment]
You might want to check the path of to the template.yaml.

Based on your provided template.yaml, the template.yaml should be in the same folder level as the target folder.

**Comment by snorith** [comment]
I had this error too. Based on the latest guide, you don't use MicronautRequestHandler as the Template specified handler anymore, you use 

`Handler: example.RoundHandler::handleRequest`

Where `example.RoundHandler` is the package and class name and `handleRequest` is the method that should be called.

When I made this update I was able to get _sam_ to test the function locally.

See: [https://micronaut-projects.github.io/micronaut-aws/latest/guide/](https://micronaut-projects.github.io/micronaut-aws/latest/guide/)
